### PR TITLE
Use glog.*f when a format string is passed

### PR DESCRIPTION
**Pull request by CaoShuFeng**
ref:
https://godoc.org/github.com/golang/glog

I use the following commands to search all the invalid usage:
$ grep "glog.Warning(" * -r | grep %
$ grep "glog.Info(" * -r | grep %
$ grep "glog.Error(" * -r | grep %
$ grep ").Info(" * -r | grep % | grep glog.V(

**Release note**:

```
NONE
```

### CODE REVIEW

**Comment by colhom** [comment]
@CaoShuFeng thanks for the PR! We also have some additional occurrences this problem in the `federation/` folder that aren't currently picked up by your regex pattern. Example:

```
glog.V(2).Info("Could not load clientConfig from %s: %v", kubeconfig, err)
```
I used: `grep 'glog\.V([0-9]).Info(' -r federation | grep %` in this case.

**Comment by colhom** [comment]
@CaoShuFeng nevermind- I went through you're entire list again and it seems like we got them all.

**Review by colhom — APPROVED** [review]

**Comment by CaoShuFeng** [comment]
/assign @lavalamp @quinton-hoole @timstclair

needs more approves.

**Comment by timstclair** [comment]
Thanks!

/lgtm

@k8s-bot ok to test

**Comment by CaoShuFeng** [comment]
rebased

**Comment by CaoShuFeng** [comment]
rebased again.

**Comment by timstclair** [comment]
/lgtm

We should consider adding a test for this... I thought gofmt picked this up?

**Comment by colhom** [comment]
@timstclair `go vet` does check a bunch of Printf-like functions, which should cover `glog.Errorf` case.

https://golang.org/cmd/vet/#hdr-Printf_family

Unfortunately `Warningf` and `Infof` aren't on that list, which I'm guessing is why this problem didn't ding any bells.

**Comment by CaoShuFeng** [comment]
rebased again.

**Comment by CaoShuFeng** [comment]
@sttts 
Hi can you help me approve this trivial fix?
I have rebased it several times.

**Comment by sttts** [comment]
@CaoShuFeng please rebase to latest master which has fixes.

File: pkg/cloudprovider/providers/aws/aws.go [hunk 1]
```diff
@@ -2201,7 +2201,7 @@ func (c *Cloud) addSecurityGroupIngress(securityGroupID string, addPermissions [
 	request.IpPermissions = changes
 	_, err = c.ec2.AuthorizeSecurityGroupIngress(request)
 	if err != nil {
-		glog.Warning("Error authorizing security group ingress %q", err)
```
Comment (CaoShuFeng) [inline pkg/cloudprovider/providers/aws/aws.go]: This is a new added one when I rebase in 2017-07-10

**Comment by CaoShuFeng** [comment]
@sttts 
Done.

File: pkg/cloudprovider/providers/aws/aws.go (hunk 1 above)
Comment (sttts) [inline pkg/cloudprovider/providers/aws/aws.go]: We need a test.

Comment (sttts) [inline pkg/cloudprovider/providers/aws/aws.go]: What about a go vet/lint upstream PR adding glog methods?

Comment (CaoShuFeng) [inline pkg/cloudprovider/providers/aws/aws.go]: colhom also thinks so.
https://github.com/kubernetes/kubernetes/pull/47829#issuecomment-312114365

**Pull request merged**